pkg/apis/streaming/v1alpha1/pulsarprovider_types.go

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+
+	"github.com/projectriff/system/pkg/apis"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+var (
+	PulsarProviderLabelKey            = GroupVersion.Group + "/pulsar-provider"             // Identifies all resources originating from a provider
+	PulsarProviderGatewayLabelKey     = GroupVersion.Group + "/pulsar-provider-gateway"     // Used as a selector
+	PulsarProviderProvisionerLabelKey = GroupVersion.Group + "/pulsar-provider-provisioner" // Used as a selector
+	PulsarProvisioner                 = "pulsar-provisioner"
+)
+
+var (
+	_ apis.Resource = (*PulsarProvider)(nil)
+)
+
+// PulsarProviderSpec defines the desired state of PulsarProvider
+type PulsarProviderSpec struct {
+	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+
+	// ServiceURL is the Pulsar URL to connect to, in the form pulsar://host:port[,host2:port2].
+	ServiceURL string `json:"serviceURL"`
```

Comment:
What does 'Service' mean in this case? If it's not a common terms for pulsar, can we find a different term to avoid confusion with k8s services?

The go client refers to the value as `URL`.

Reply:
It's used throughout the [Pulsar doc](https://pulsar.apache.org/docs/en/reference-configuration/) as well as by liiklus. Please note I did not use/include the term "broker" on purpose, as a broker in Pulsar has a particular meaning (a "cluster" can be composed of several "brokers")